student-info-system/student/views.py

```diff
@@ -139,33 +139,44 @@ def history(request):
     data = {
         'auser': the_user,
     }
-    data.update(
-        filtered_table2(
-            name='history',
-            qs=the_user.profile.student.course_history(),
-            filter=StudentHistoryFilter,
-            table=StudentHistoryTable,
-            request=request,
-            wrap_list=False,
-            self_url=reverse('student:history'),
-            click_url=reverse('schooladmin:sectionstudent', args=[DUMMY_ID]),
-        ))
-    data.update(
-        filtered_table2(
-            name='semesters',
-            qs=the_user.profile.student.semesters.all(),
-            filter=SemesterFilter,
-            table=SemestersSummaryTable,
-            request=request,
-            self_url=reverse('student:history'),
-            click_url=reverse('schooladmin:semester', args=[DUMMY_ID]),
-        ))
+    ssects = ssects_by_sem(the_user)
```

Comment:
seems like something best done in the model. Take a look at Student.gpa. this is almost the same code...

Reply:
done